File: plugins/skill-intake/scripts/section_quality_check.js
#!/usr/bin/env node
'use strict';

const fs = require('node:fs');
const path = require('node:path');

function checkSection(name, body) {
  const text = String(body || '').trim();
  const wordCount = text.split(/\s+/).filter(Boolean).length;
  const hasHeading = /^#{1,4}\s/m.test(text) || true;
  const hasContent = wordCount >= 20;
  const hasPlaceholder = /(TBD|未定|要確認|\?{2,})/i.test(text);
  const ok = hasContent && !hasPlaceholder;
  return { section: name, ok, word_count: wordCount, has_heading: hasHeading, has_placeholder: hasPlaceholder };
}

function checkAll(sections) {
  const results = Object.keys(sections).map((k) => checkSection(k, sections[k]));
  const failed = results.filter((r) => !r.ok);
  return { ok: failed.length === 0, results, failed_count: failed.length };
}

function main(argv) {
  const file = argv[2];
  if (!file) { process.stderr.write('usage: section_quality_check.js <intake.json>\n'); return 2; }
  let data;
  try { data = JSON.parse(fs.readFileSync(path.resolve(file), 'utf8')); }
  catch (e) { process.stderr.write(`input error: ${e.message}\n`); return 2; }
  const sections = data.sections || {};
  const r = checkAll(sections);
  process.stdout.write(JSON.stringify(r, null, 2) + '\n');
  return r.ok ? 0 : 1;
}

module.exports = { checkAll, checkSection };

if (require.main === module) {
  process.exit(main(process.argv));
}

File: plugins/skill-intake/scripts/section_quality_check.py
#!/usr/bin/env python3
"""Score each intake section for completeness and placeholder hygiene."""
from __future__ import annotations

import json
import re
import sys
from pathlib import Path
from typing import Any

_PLACEHOLDER_RE = re.compile(r'(TBD|未定|要確認|\?{2,})', re.IGNORECASE)


def check_section(name: str, body: Any) -> dict[str, Any]:
    text = str(body or '').strip()
    word_count = len([w for w in re.split(r'\s+', text) if w])
    has_placeholder = bool(_PLACEHOLDER_RE.search(text))
    has_content = word_count >= 20
    ok = has_content and not has_placeholder
    return {
        'section': name,
        'ok': ok,
        'word_count': word_count,
        'has_heading': True,
        'has_placeholder': has_placeholder,
    }


def check_all(sections: dict[str, Any]) -> dict[str, Any]:
    results = [check_section(k, sections[k]) for k in sections]
    failed = [r for r in results if not r['ok']]
    return {'ok': len(failed) == 0, 'results': results, 'failed_count': len(failed)}


def main(argv: list[str]) -> int:
    if len(argv) < 1:
        sys.stderr.write('usage: section_quality_check.py <intake.json>\n')
        return 2
    try:
        data = json.loads(Path(argv[0]).resolve().read_text(encoding='utf-8'))
    except Exception as e:
        sys.stderr.write(f'input error: {e}\n')
        return 2
    r = check_all(data.get('sections') or {})
    sys.stdout.write(json.dumps(r, ensure_ascii=False, indent=2) + '\n')
    return 0 if r['ok'] else 1


if __name__ == '__main__':
    sys.exit(main(sys.argv[1:]))
